### PR TITLE
Fixed some Sonar complaints

### PR DESCRIPTION
**Pull request by NiceGuyAndersson**

### CODE REVIEW

File: src/Engine/Audio/SteamAudio.cpp [hunk 1]
```diff
@@ -45,8 +46,8 @@ void SteamAudio::GetFinalMix(IPLAudioBuffer* finalBuf, uint32_t* numSamples) {
 
     // Clean up
     for (IPLAudioBuffer buf : processedBuffers) {
```
Comment (NinjaDanz3r) [inline src/Engine/Audio/SteamAudio.cpp]: INFO keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

File: src/Engine/Audio/SteamAudio.cpp [hunk 2]
```diff
@@ -1,15 +1,16 @@
 #include "SteamAudio.hpp"
 
 SteamAudio::SteamAudio() {
-
+    context = nullptr;
+    environmentalRenderer = nullptr;
 }
```
Comment (NinjaDanz3r) [inline src/Engine/Audio/SteamAudio.cpp]: MINOR Member variable 'SteamAudio::context' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'SteamAudio::environmentalRenderer' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

File: src/Engine/Audio/SteamAudioInterface.cpp [hunk 3]
```diff
@@ -2,11 +2,18 @@
 
 
 SteamAudioInterface::SteamAudioInterface() {
```
Comment (NinjaDanz3r) [inline src/Engine/Audio/SteamAudioInterface.cpp]: MINOR Member variable 'SteamAudioInterface::context' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'SteamAudioInterface::playerPos' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'SteamAudioInterface::playerDir' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
MINOR Member variable 'SteamAudioInterface::playerUp' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

File: src/Engine/Audio/SteamAudio.hpp [hunk 4]
```diff
@@ -37,8 +37,6 @@ class SteamAudio {
          */
         void GetFinalMix(IPLAudioBuffer* finalBuf, uint32_t* numSamples);
 
-        
-
     private:
```
Comment (NinjaDanz3r) [inline src/Engine/Audio/SteamAudio.hpp]: INFO too many consecutive empty lines [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AL005)

File: src/Engine/Audio/SteamAudioIndirectRenderer.cpp [hunk 5]
```diff
@@ -1,11 +1,13 @@
 #include "SteamAudioIndirectRenderer.hpp"
 
 SteamAudioIndirectRenderer::SteamAudioIndirectRenderer() {
-
+    environment = nullptr;
+    context = nullptr;
 }
```
Comment (NinjaDanz3r) [inline src/Engine/Audio/SteamAudioIndirectRenderer.cpp]: MINOR Member variable 'SteamAudioIndirectRenderer::environment' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

File: src/Engine/Audio/SteamAudioDirectRenderer.cpp [hunk 6]
```diff
@@ -1,11 +1,13 @@
 #include "SteamAudioDirectRenderer.hpp"
 
 SteamAudioDirectRenderer::SteamAudioDirectRenderer() {
-
+    context = nullptr;
+    environment = nullptr;
 }
```
Comment (NinjaDanz3r) [inline src/Engine/Audio/SteamAudioDirectRenderer.cpp]: MINOR Member variable 'SteamAudioDirectRenderer::environment' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

File: src/Engine/Manager/SoundManager.cpp [hunk 7]
```diff
@@ -77,7 +77,7 @@ void SoundManager::Update(float deltaTime) {
 
             float* soundBuf = new float[numSamples];
             if (sound->soundBuffer->GetSize() > sound->place + numSamples) {
-                std::memcpy(soundBuf,(sound->soundBuffer->GetBuffer() + sound->place), sizeof(float)*numSamples);
+                std::memcpy(soundBuf, (sound->soundBuffer->GetBuffer() + sound->place), sizeof(float)*numSamples);
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/SoundManager.cpp]: INFO comma should be followed by whitespace [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)

File: src/Engine/Manager/SoundManager.cpp [hunk 8]
```diff
@@ -116,7 +116,7 @@ void SoundManager::Update(float deltaTime) {
         processedSamples = new float[numSamples] {0};
 
     Pa_WriteStream(stream, processedSamples, *numProcessedSamples);
-    delete processedSamples;
+    delete[] processedSamples;
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/SoundManager.cpp]: MAJOR Mismatching allocation and deallocation: processedSamples [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmismatchAllocDealloc)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [VorbisFile.cpp#L8](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/06f76df57adeb23c9ca98e8af7ed9d3f2c3da50a/src/Engine/Audio/VorbisFile.cpp#L8): Member variable 'VorbisFile::sampleRate' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [SoundManager.cpp#L153](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/06f76df57adeb23c9ca98e8af7ed9d3f2c3da50a/src/Engine/Manager/SoundManager.cpp#L153): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Audio/SteamAudio.cpp [hunk 9]
```diff
@@ -1,15 +1,16 @@
...
 
 SteamAudio::SteamAudio(IPLContext * context, IPLhandle * environment) {
 
 }
 
 SteamAudio::~SteamAudio() {
-
+    delete context;
```
Comment (Chainsawkitten) [inline src/Engine/Audio/SteamAudio.cpp]: Is context guaranteed not to be nullptr? (Since it's set to nullptr in constructor.)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [VorbisFile.cpp#L8](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e66f34837911a3ae83ac123e27f5d96b7abde831/src/Engine/Audio/VorbisFile.cpp#L8): Member variable 'VorbisFile::sampleRate' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [SoundManager.cpp#L153](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e66f34837911a3ae83ac123e27f5d96b7abde831/src/Engine/Manager/SoundManager.cpp#L153): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
iamheman

File: src/Engine/Audio/SteamAudio.cpp [hunk 10]
```diff
@@ -45,8 +47,8 @@ void SteamAudio::GetFinalMix(IPLAudioBuffer* finalBuf, uint32_t* numSamples) {
 
     // Clean up
```
Comment (NinjaDanz3r) [inline src/Engine/Audio/SteamAudio.cpp]: INFO keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [VorbisFile.cpp#L8](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d0e276ee7c34f733a68f4aaf4ac963ae61234f35/src/Engine/Audio/VorbisFile.cpp#L8): Member variable 'VorbisFile::sampleRate' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [SoundManager.cpp#L153](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d0e276ee7c34f733a68f4aaf4ac963ae61234f35/src/Engine/Manager/SoundManager.cpp#L153): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [VorbisFile.cpp#L8](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/868ff8f5ab6476b9e8ef07bee510145895b79ccb/src/Engine/Audio/VorbisFile.cpp#L8): Member variable 'VorbisFile::sampleRate' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)
1. ![MINOR][MINOR] [SoundManager.cpp#L158](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/868ff8f5ab6476b9e8ef07bee510145895b79ccb/src/Engine/Manager/SoundManager.cpp#L158): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**